### PR TITLE
Pull request template and coding style md

### PR DESCRIPTION
**Pull request by ixTec**
# Add pull request template and coding style document

Please delete options that are not relevant.

- `docs`: Changes consisting of documentation update(s)
- `chore`: Changes that do not affect production code output

# Changes

## ❓ Misc Changes

- Added pull request template
- Added initial CODING_STYLES.md

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+- `docs`: Changes consisting of documentation update(s)
+- `perf`: Changes that improve performance
+- `test`: Updates to internal component / utility tests
+- `ci`: Changes to the CI configuration / scripts
+- `build`: Changes to build configuration, irrelevant to the user
+- `refactor`: Changes to the output that neither fix an issue nor add a feature
+- `chore`: Changes that do not affect production code output
+- `BREAKING-CHANGE`: Breaking change (fix or feature that would cause existing functionality to no longer work as expected)
+
+# Changes
```
Comment (blurbyte) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: Really like this section.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Insert Description Here
+
+## [WF-\*](https://datacamp.atlassian.net/browse/WF-*)
+
+Please delete options that are not relevant.
```
Comment (blurbyte) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: Not really sure what the purpose of this list is.

Comment (blurbyte) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: Not really sure what the purpose of this list is.

File: docs/CODING_STYLE.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+🚧 Please note, these are still under construction and so may change as **Waffles** grows.
+
+## Guidelines
+
+1. Use kebab-case for file names.
+2. Use regular fuction declaration (`function() {}`), including for functional components
+3. Use arrow function declaration (`() => {}`) for anonymous functions.
+4. For targetting components in tests directly (both e2e and unit), use the `data-testid` property. E.g. `<Button data-testid="example-button" ... >Example Button</Button>`
+5. If using `TODO` comments, add your GitHub handle alongside the `@` prefix. E.g. `// TODO Implement important feature @handle`
+6. Separate internal and external imports into blocks with a new line.
```
Comment (blurbyte) [inline docs/CODING_STYLE.md:14]: Hopefully linter can take care of it.

**Review by blurbyte — APPROVED** [review]
Looks nice! Just not sure about a list of commit tags at the beginning of PR template. What's the purpose of it? I also found headings a bit too big besides first one (would be a bit more semantically correct to have only 1 H1 too).

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (ixTec) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: The list is just to provide an overview of the type of changes in the pull request (e.g. if it's a fix, feat or breaking change then we may need to do more testing / review before merging in comparison to something like a chore or style change

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (ixTec) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: This is more for a overall heading for the below sections, so we could make this h2 and the children h3 as a result?

Comment (ixTec) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: This is more for a overall heading for the below sections, so we could make this h2 and the children h3 as a result?

**Pull request merged**